### PR TITLE
forum security

### PR DESCRIPTION
**Pull request by LEEDASILVA**

### CODE REVIEW

File: subjects/forum/forum-security.en.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+- You should encrypt :
+  - Clients passwords
+  - Database
+  - Clients session cookies should be unique. For instance, the session state is stored on the server and the session should present an unique identifier. This way the client has no direct access to it. Therefore, there is no way for attackers to read or tamper with session state.
+
+This project will help you learn about :
+
+- HTTPS
+- Encryption
+  - Database
```
Comment (xpetit) [inline subjects/forum/forum-security.en.md]: Same question

File: subjects/forum/forum-security.en.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  - password
+  - session/cookies
+  - Universal Unique Identifier (UUI)
```
Comment (xpetit) [inline subjects/forum/forum-security.en.md]: UUID*

File: subjects/forum/forum-security.en.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+### Hints
+
+- You can take a look at the `openssl` manual.
+- For the session cookies you can take a look at the [Universal Unique Identifier (UUI)](https://en.wikipedia.org/wiki/Universally_unique_identifier)
```
Comment (xpetit) [inline subjects/forum/forum-security.en.md]: UUID*

File: subjects/forum/forum-security.audit.en.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#### Functional
+
+##### Try opening the forum.
+###### Does the URL contain HTTPS?
+
+###### Is the Go TLS structure well configured?
+
+###### Is the database encrypt?
```
[the remark this answers is not in the document]
Comment (LEEDASILVA) [reply subjects/forum/forum-security.audit.en.md]: For the database i was thinking of adding a password to it. About the encryption for the channel communication i'm not sure how it works in go. Will take a look thanks for the idea

File: subjects/forum/forum-security.audit.en.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+##### Try to login into the forum and open the inspector(CTRL+SHIFT+i) and go to the storage to see the cookies(this can be different depending on the [browser](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools)).
+###### Does the session cookie present a unique identifier?
+
+#### General
+
+###### +Does the project implement their own certificates for the HTTPS protocol?
+
+###### +Does the project implement UUI(Universal Unique Identifier) for the user session?
+
+###### +Does the project implement [environment variables](https://en.wikipedia.org/wiki/Environment_variable) (.env file), for the TLS certificates?
```
[the remark this answers is not in the document]
Comment (LEEDASILVA) [reply subjects/forum/forum-security.audit.en.md]: right thanks Xav :)

File: subjects/forum/forum-security.audit.en.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (xpetit) [reply subjects/forum/forum-security.audit.en.md]: Is it possible to encrypt the data with SQLite ?
Usually a password is to connect to the database, not to encrypt the data

[the remark this answers is not in the document]
Comment (LEEDASILVA) [reply subjects/forum/forum-security.audit.en.md]: yes the password is for the connection, the encryption of the data i'm still reading the sqlite3 doc

[the remark this answers is not in the document]
Comment (xpetit) [reply subjects/forum/forum-security.audit.en.md]: Ok then please specify it is the connection to the database which is encrypted, not the database itself ^^

[the remark this answers is not in the document]
Comment (LEEDASILVA) [reply subjects/forum/forum-security.audit.en.md]: ok :) thanks

**Pull request merged**